client/layout/style.scss

```diff
@@ -423,3 +423,96 @@ body.is-mobile-app-view {
 		}
 	}
 }
+
+// Handle Gravatar domain flow styles
+.layout.is-domain-for-gravatar {
+	* {
+		font-family:
+			'SF Pro Text',
+			-apple-system,
+			BlinkMacSystemFont,
+			'Segoe UI',
+			Roboto,
+			Oxygen-Sans,
+			Ubuntu,
+			Cantarell,
+			'Helvetica Neue',
+			sans-serif !important;
+	}
+
+	.masterbar__secure-checkout {
+		column-gap: 8px;
+	}
+
+	.gravatar-text-logo {
+		width: 119px;
+		height: 22px;
+	}
+
+	h1.formatted-header__title,
+	.checkout-title,
+	h1.thank-you__header-title {
+		font-family:
+			'SF Pro Display',
+			-apple-system,
+			BlinkMacSystemFont,
+			'Segoe UI',
+			Roboto,
+			Oxygen-Sans,
+			Ubuntu,
+			Cantarell,
+			'Helvetica Neue',
+			sans-serif !important;
+	}
+
+	h1.formatted-header__title,
+	.checkout-title,
+	.wp-checkout-order-summary__label,
+	.wp-checkout-order-summary__total-price,
```

Comment:
Non-blocking, but you can drop these by extending the `checkoutTheme.weights` variable in `client/my-sites/checkout/src/components/checkout-main.tsx` (similar to how you did `gravatarColors`). It's not as relevant now, but we're working on separating Checkout from Calypso, and this would make sure that the styling persists.